src/main/java/org/springframework/integration/kafka/core/DefaultConnectionFactory.java

```diff
@@ -50,28 +51,21 @@
  */
 public class DefaultConnectionFactory implements InitializingBean, ConnectionFactory, DisposableBean {
 
+	private final static Log log = LogFactory.getLog(DefaultConnectionFactory.class);
+
+	public static final Predicate<TopicMetadata> errorlessTopicMetadataPredicate = new ErrorlessTopicMetadataPredicate();
+
 	private final GetBrokersByPartitionFunction getBrokersByPartitionFunction = new GetBrokersByPartitionFunction();
 
 	private final ConnectionInstantiationFunction connectionInstantiationFunction = new ConnectionInstantiationFunction();
 
 	private final Configuration configuration;
 
-	private final AtomicReference<PartitionBrokerMap> partitionBrokerMapReference = new AtomicReference<PartitionBrokerMap>();
+	private final AtomicReference<MetadataCache> metadataCacheHolder =
```

Comment:
`volatile` could work as well, eventually. I think it communicates better the intent of being a data holder, though.